packages/sprite-animated/test/AnimatedSprite.js

```diff
@@ -95,4 +95,214 @@ describe('PIXI.AnimatedSprite', function ()
+            {
+                expect(frameIds).to.deep.equal([1, 2]);
+                expect(this.sprite.playing).to.be.false;
+                this.sprite.onComplete = null;
+                this.sprite.onFrameChange = null;
+                done();
+            };
+            this.sprite.onFrameChange = (frame) =>
+            {
+                frameIds.push(frame);
+            };
+            this.sprite.gotoAndPlay(1);
+            expect(this.sprite.playing).to.be.true;
+        });
+    });
+
+    describe('.gotoAndStop()', function ()
+    {
+        before(function ()
+        {
+            this.sprite = new AnimatedSprite([Texture.EMPTY, Texture.EMPTY, Texture.EMPTY]);
+            this.sprite.animationSpeed = 1;
+            this.sprite.loop = false;
+        });
+
+        after(function ()
+        {
+            this.sprite.destroy();
+            this.sprite = null;
+        });
+
+        beforeEach(function ()
+        {
+            this.sprite.playing = false;
+        });
+
+        it('should fire onFrameChange on target frame', function (done)
+        {
+            const targetFrame = 1;
+
+            this.sprite.animationSpeed = 1;
+            this.sprite.onFrameChange = (frame) =>
+            {
+                expect(frame).to.equal(targetFrame);
+                expect(this.sprite.playing).to.be.false;
+                this.sprite.onComplete = null;
+                this.sprite.onFrameChange = null;
+                done();
+            };
+            this.sprite.gotoAndStop(targetFrame);
+            expect(this.sprite.playing).to.be.false;
+        });
+
+        it('should not fire onFrameChange on target frame if current is already target', function ()
+        {
+            let fired = false;
+            const targetFrame = 1;
+
+            this.sprite.gotoAndStop(targetFrame);
+
+            this.sprite.animationSpeed = 1;
+            this.sprite.onFrameChange = () =>
+            {
+                fired = true;
+            };
+            this.sprite.gotoAndStop(targetFrame);
+            expect(this.sprite.playing).to.be.false;
+            expect(fired).to.be.false;
+        });
+    });
+
+    describe('.onFrameChange()', function ()
+    {
+        before(function ()
+        {
+            this.sprite = new AnimatedSprite([Texture.EMPTY, Texture.EMPTY, Texture.EMPTY]);
+            this.sprite.animationSpeed = 1;
+            this.sprite.loop = false;
+        });
+
+        after(function ()
+        {
+            this.sprite.destroy();
+            this.sprite = null;
+        });
+
+        beforeEach(function ()
+        {
+            this.sprite.playing = false;
+        });
+
+        it('should fire every frame(except current) during one play', function (done)
+        {
+            const frameIds = [];
+
+            this.sprite.gotoAndStop(0);
+            this.sprite.animationSpeed = 1;
+            this.sprite.onComplete = () =>
+            {
+                done();
```

Comment:
Might be better to call done after all the other expect statements?

Reply:
yes, my stupid mistake.